contracts/core/MarginPoolV2.sol

```diff
@@ -0,0 +1,244 @@
+/**
+ * SPDX-License-Identifier: UNLICENSED
+ */
+pragma solidity =0.6.10;
+
+import {ERC20Interface} from "../interfaces/ERC20Interface.sol";
+import {OtokenInterface} from "../interfaces/OtokenInterface.sol";
+import {AddressBookInterface} from "../interfaces/AddressBookInterface.sol";
+import {WhitelistInterface} from "../interfaces/WhitelistInterface.sol";
+import {SafeMath} from "../packages/oz/SafeMath.sol";
+import {SafeERC20} from "../packages/oz/SafeERC20.sol";
+import {MarginPool} from "./MarginPool.sol";
+
+/**
+ * @author Ribbon Team
+ * @title MarginPoolV2
+ * @notice Contract that holds all protocol funds AND allows collateral borrows
+ */
+contract MarginPoolV2 is MarginPool {
+    using SafeMath for uint256;
+    using SafeERC20 for ERC20Interface;
+
+    uint16 public constant TOTAL_PCT = 10000; // Equals 100%
+
+    /// @dev mapping between collateral asset and borrow PCT
+    mapping(address => uint256) public borrowPCT;
+    /// @dev mapping between address and whitelist status of borrower
+    mapping(address => bool) public whitelistedBorrower;
+    /// @dev mapping between address and whitelist status of otoken buyer
+    /// This is the whitelist for all the holders of oTokens that have a claim to
+    /// collateral in this pool
+    mapping(address => bool) public whitelistedOTokenBuyer;
+    /// @dev mapping between address and whitelist status of ribbon vault
+    mapping(address => bool) public whitelistedRibbonVault;
+    /// @dev mapping between (borrower, asset) and outstanding borrower amount
```

Comment:
mappings that are crucial for knowing whether to direct vaults to margin pool v1 or v2

whitelistedOTokenBuyer: knowing which margin pool to claim an oToken from
whitelistedRibbonVault: knowing which margin pool to set a margin vault to